src/main/java/com/bynder/sdk/api/ApiFactory.java

```diff
@@ -119,96 +110,143 @@ private static OkHttpClient createOkHttpClient(final Configuration configuration
      * @param httpClientBuilder Builder instance of the HTTP client.
      * @param configuration {@link Configuration} settings for the HTTP communication with Bynder.
      */
-    private static void setOAuthInterceptor(final Builder httpClientBuilder,
-        final Configuration configuration) {
-        httpClientBuilder.addInterceptor(new Interceptor() {
-
-            @Override
-            public Response intercept(final Chain chain) throws IOException {
-                if (configuration.getOAuthSettings().getToken() == null) {
-                    throw new BynderRuntimeException("Token is not defined in Configuration");
-                }
-
-                // check if access token is expiring in the next 15 seconds
-                if (Utils.isDateExpiring(configuration.getOAuthSettings().getToken().getAccessTokenExpiration(), 15)) {
-                    // refresh the access token
-                    OAuthService oAuthService = BynderClient.Builder.create(configuration)
-                        .getOAuthService();
-                    Token token = oAuthService.refreshAccessToken().blockingSingle();
-
-                    // trigger callback method
-                    configuration.getOAuthSettings().callback(token);
-                }
-
-                String headerValue = String
-                    .format("%s %s", "Bearer", configuration.getOAuthSettings().getToken().getAccessToken());
-
-                Request.Builder requestBuilder = chain.request().newBuilder()
-                    .header("Authorization", headerValue);
-
-                Request request = requestBuilder.build();
-                return chain.proceed(request);
+    private static void setOAuthInterceptor(
+            final Builder httpClientBuilder,
+            final Configuration configuration
+    ) {
+        httpClientBuilder.addInterceptor(chain -> {
+            if (configuration.getOAuthSettings().getToken() == null) {
+                throw new BynderRuntimeException("Token is not defined in Configuration");
+            }
```

Comment:
This is clean. Perhaps we should null check `configuration` before this line to also throw a `BynderRuntimeException` ?
Because line 118 might not be executed if `configuration` is null, it will just throw a NPE which will propagate through the code.

Reply:
I didn't really change the logic here, just the code style and reordered certain things. But besides `configuration` can't be null in practice, for example intellij also makes this assumption based on the rest of the code.

I can imagine `configuration.getOAuthSettings().getToken() == null` could happen with some missconfig or something maybe, but not sure about the reasoning behind this, when exactly this could occur.

Reply:
I see, that's why I just added a comment to check with you guys what you thought about it. Thanks for the clarification.